### PR TITLE
Avoid clearing cache for every dataset

### PR DESCRIPTION
**Pull request by phofl**
This get's rid of most of the overhead that we can avoid

### CODE REVIEW

File: dask_expr/io/parquet.py [hunk 1]
```diff
@@ -55,6 +55,12 @@
 _cached_plan = {}
 
 
+def _control_cached_dataset_info(key):
+    if len(_cached_dataset_info) > 10 and key not in _cached_dataset_info:
+        key_to_pop = list(_cached_dataset_info.keys())[0]
+        _cached_dataset_info.pop(key_to_pop)
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/io/parquet.py:61]: ``_cached_plan`` seems to be fine, variable makes sense I think

File: dask_expr/io/parquet.py [hunk 2]
```diff
@@ -50,11 +50,21 @@
 
 NONE_LABEL = "__null_dask_index__"
 
-# TODO: Allow _cached_dataset_info/_plan to contain >1 item?
 _cached_dataset_info = {}
+_CACHED_DATASET_SIZE = 10
+# TODO: Allow _cached__plan to contain >1 item?
```
Comment (rjzamora) [inline dask_expr/io/parquet.py:55]:
```suggestion
# TODO: Allow _cached_plan to contain >1 item?
```
Minor typo

**Review by rjzamora — APPROVED** [review]

**Comment by mrocklin** [comment]
I'm curious, how does this affect the query optimization time in your
recent benchmarks?

**Comment by phofl** [comment]
It got us down to the same ballpark as dask/dask, so we don't waste unnecessary time anymore. It still takes some time because we have to read from remote storage

**Pull request merged**